pkg/hubble/parser/common/labels.go

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package common
+
+import (
+	"net"
+	"sort"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/cilium/cilium/pkg/identity"
+)
+
+func FilterCIDRLabels(log logrus.FieldLogger, labels []string) []string {
+	// Cilium might return a bunch of cidr labels with different prefix length. Filter out all
+	// but the longest prefix cidr label, which can be useful for troubleshooting. This also
+	// relies on the fact that when a Cilium security identity has multiple CIDR labels, longer
+	// prefix is always a subset of shorter prefix.
+	cidrPrefix := "cidr:"
+	var filteredLabels []string
+	var max *net.IPNet
+	var maxStr string
+	for _, label := range labels {
+		if strings.HasPrefix(label, cidrPrefix) {
+			currLabel := strings.TrimPrefix(label, cidrPrefix)
+			// labels for IPv6 addresses are represented with - instead of : as
+			// : cannot be used in labels; make sure to convert it to a valid
+			// IPv6 representation
+			currLabel = strings.Replace(currLabel, "-", ":", -1)
+			_, curr, err := net.ParseCIDR(currLabel)
+			if err != nil {
+				log.WithField("label", label).Warn("got an invalid cidr label")
+				continue
+			}
+			if max == nil {
+				max = curr
+				maxStr = label
+			}
+			currMask, _ := curr.Mask.Size()
+			maxMask, _ := max.Mask.Size()
+			if currMask > maxMask {
+				max = curr
+				maxStr = label
+			}
```

Comment:
nit: IIUC technically we don't need to remember the actual max `IPNet` here, only its mask size. Assuming `var max *net.IPNet` was replaced by `var maxSize int` earlier we could

```suggestion
			if currMask, _ := curr.Mask.Size(); currMask > maxSize {
				maxSize, maxStr = currMask, label
			}
```

This simplify the logic a bit and also avoid recomputing the max mask size.